tests/foreman/sys/test_custom_cert_installation.py

```diff
@@ -0,0 +1,120 @@
+    """Update the satellite instance with new custom certs.
+
+    :id: 6408226e-6c7f-4394-8a60-fb86f058f3fd
+
+    :steps:
+        1. Generate the custom certs
+        2. Update satellite with custom certs
+        3. Assert output does not report SSL certificate error
+        4. Assert all services are running
+
+    :expectedresults: Satellite should be updated with custom certs
+    """
+    satellite = satellite_latest.hostname
+    cert_data = custom_certs_factory(hostname=satellite)
+    with get_connection(hostname=satellite) as connection:
+        result = connection.run(
+            'katello-certs-check -c {} -k {} -b {}'.format(
+                cert_data['cert_file_name'],
+                cert_data['key_file_name'],
+                cert_data['ca_bundle_file_name'],
+            ),
+            output_format='plain',
+        )
+        assert result.return_code == 0, 'Failed to generate the custom cert check command'
+
+        # update the custom certs in Satellite
+        command = f'satellite-installer --scenario satellite \
+                          --certs-server-cert "/root/{satellite}/{satellite}.crt" \
+                          --certs-server-key "/root/{satellite}/{satellite}.key" \
+                          --certs-server-ca-cert "/root/cacert.crt" \
+                          --certs-update-server --certs-update-server-ca'
+        result = connection.run(command, output_format='plain', timeout=1500)
+        assert result.return_code == 0, 'Failed to update the satellite with custom certs'
+        # assert no hammer ping SSL cert error
+        result = connection.run('hammer ping')
+        assert 'SSL certificate verification failed' not in result.stdout
+        assert 'ok' in result.stdout[1]
+        # assert all services are running
+        result = connection.run('foreman-maintain health check --label services-up -y')
+        assert result.return_code == 0, 'Not all services are running'
+
+
+@pytest.mark.parametrize(
+    'param_host', [{'target_memory': '10GiB', 'nick': 'rhel7'}], ids=['rhel7'], indirect=True
+)
+@pytest.mark.tier1
+def test_custom_cert_install_satellite(param_host, custom_certs_factory):
+    """Install the satellite instance with new custom certs.
+
+    :id: 1580564c-a714-4cd7-a80f-0e635dcec78e
+
+    :steps:
+        1. Generate the custom certs
+        2. Install satellite with custom certs
+        3. Assert output does not report SSL certificate error
+        4. Assert all services are running
+
+    :expectedresults: Satellite should be updated with custom certs
+    """
+    rhel_host = param_host.hostname
+    custom_certs_factory(hostname=rhel_host)
+    with get_connection(hostname=rhel_host) as connection:
+        # subscribe satellite dog_food server
+        dogfood_url = (
+            f'{settings.repos.dogfood_repo_host}/pub/katello-ca-consumer-latest.noarch.rpm'
+        )
+        activation_key = f'satellite-{settings.robottelo.satellite_version}-qa-rhel7'
```

Comment:
The activation key `rhel7` component is going to be dependent on the content host that `param_host` fixture provided. You need to resolve it dynamically.